Client/Frontend/Settings/SearchBar/SearchBarSettingsViewModel.swift

```diff
@@ -59,10 +59,10 @@ final class SearchBarSettingsViewModel: FeatureFlaggable {
     weak var delegate: SearchBarPreferenceDelegate?
 
     private let prefs: Prefs
-    private let notificationCenter: NotificationCenter
-    init(prefs: Prefs, notificationCenter: NotificationCenter = NotificationCenter.default) {
+    private let notificationProtocol: NotificationProtocol
```

Comment:
We can keep the name of `notificationCenter` even thought the type is now of `NotificationProtocol`